### PR TITLE
Added error charts

### PR DESCRIPTION
**Pull request by jryu01**
# Description

Screenshot 2024-07-29 at 9 55 35 AM

Resolves #(issue)

### CODE REVIEW

File: packages/client/hmi-client/src/services/charts.ts [hunk 1]
```diff
@@ -32,6 +32,185 @@ export interface HistogramChartOptions extends BaseChartOptions {
...
+	areaChartHeight?: number;
+	boxPlotHeight?: number;
+	variables: { field: string; label?: string }[];
+}
+
+export const createErrorChart = (dataset: Record<string, any>[], options: ErrorChartOptions) => {
+	const axisColor = '#EEE';
+	const labelColor = '#667085';
+	const labelFontWeight = 'normal';
+	const globalFont = 'Figtree';
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/charts.ts:47]: Not for this PR, but we should make those constants for ALL the charts on the application.

File: packages/client/hmi-client/src/services/charts.ts [hunk 2]
```diff
@@ -32,6 +32,185 @@ export interface HistogramChartOptions extends BaseChartOptions {
...
+
+	const variables = variablesOptions.map((v) => v.field);
+	const maxbins = options.maxBins ?? 30;
+
+	const titleObj = options.title
+		? {
+				text: options.title,
+				anchor: 'start',
+				subtitle: ' ',
+				subtitlePadding: 4
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/charts.ts:73]: Why adding an empty space subtitle and a padding if there are none?

File: packages/client/hmi-client/src/temp/Vegalite.vue [hunk 3]
```diff
@@ -17,8 +16,8 @@
 import { debounce } from 'lodash';
 import { ref, onMounted } from 'vue';
 import VegaChart from '@/components/widgets/VegaChart.vue';
-import unchartedVegaTheme from './vega-theme';
-import { createForecastChart, createHistogramChart } from '@/services/charts';
+// import unchartedVegaTheme from './vega-theme';
```
Comment (YohannParis) [inline packages/client/hmi-client/src/temp/Vegalite.vue:19]:
```suggestion
```

File: packages/client/hmi-client/src/temp/Vegalite.vue [hunk 4]
```diff
@@ -37,12 +36,15 @@ for (let j = 0; j < numPoints; j++) {
 
 for (let i = 0; i < numSamples; i++) {
 	let error = 0;
+	let error2 = 0;
```
Comment (YohannParis) [inline packages/client/hmi-client/src/temp/Vegalite.vue:39]: Could you add a comment to explain the purpose of this `error2`?

**Review by YohannParis — APPROVED** [review]

File: packages/client/hmi-client/src/temp/Vegalite.vue (hunk 4 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/temp/Vegalite.vue:39]: First rule of `/temp` folder: you do not talk about `/temp` folder 😄

File: packages/client/hmi-client/src/services/charts.ts [hunk 5]
```diff
@@ -32,6 +32,177 @@ export interface HistogramChartOptions extends BaseChartOptions {
...
+			box: {
+				fill: 'transparent',
+				stroke: boxPlotColor,
+				strokeWidth: 1
+			}
+		}
+	};
+
+	return {
+		$schema: 'https://vega.github.io/schema/vega-lite/v5.json',
```
Comment (mwdchang) [inline packages/client/hmi-client/src/services/charts.ts:114]: We got a constant for this at the top of the file

**Review by mwdchang — APPROVED** [review]

**Comment by mwdchang** [comment]
Looks good, let's get this in as a base and we can continue the threads from pascale and nelson:
- whether we want to go a single chart per n-variable, or we want one-to-one relations
- check the kernel density quirkiness and x-axis alignment stuff nelson was talking about

File: packages/client/hmi-client/src/services/charts.ts (hunk 2 above)
Comment (jryu01) [reply packages/client/hmi-client/src/services/charts.ts:71]: Probably it's to add padding below the main title.

File: packages/client/hmi-client/src/services/charts.ts (hunk 1 above)
Comment (jryu01) [reply packages/client/hmi-client/src/services/charts.ts:46]: That makes sense, we can probably need to revisit this and do some clean ups

**Comment by jryu01** [comment]
> Looks good, let's get this in as a base and we can continue the threads from pascale and nelson:
> 
> * whether we want to go a single chart per n-variable, or we want one-to-one relations
> * check the kernel density quirkiness and x-axis alignment stuff nelson was talking about
Screenshot 2024-07-30 at 9 16 02 AM


Sounds good! By the way the x-alignment stuff nelson was talking about is addressed

**Pull request merged**